src/entities/Habitat.ts

```diff
@@ -0,0 +1,40 @@
+import Tile from "./Tile";
+
+/**
+ * Habitat is a child of Tile.
+ * A Habitat must:
+ * a) Be placed next to at least 1 other Tile.
+ * b) Border only tiles that share at least one habitatType with it.
+*/
+class Habitat extends Tile {
+    habitatType;
+    /**
+     * Create a new Habitat.
+     * @param {array} habitatType - Array of strings. The habitat type(s) of the object. 
+     * @param {number} sides - Number of sides the tile will have. 6 (i.e. hexagon) by default. 
+     */
+    constructor(habitatType, sides = 6) {
+        super(sides)
+        this.habitatType = habitatType
+    }
+    /**
+    * Check that all neighbours are of the allowed habitat type.
+    * 
+    * @returns Logical. Does the Tile have correct neighbours?
+    */
+    checkNeighbours() {
+        var total = 0
+        for (const neighbour of this.neighbours) {
+            // Neighbours may either be null or share at least one type with this object.
+            total += Number(!(neighbour === null || this.habitatType.some(type => neighbour.habitatType.includes(type))))
+        }
+        if (total > 0) {
+            console.log("Invalid neighbour. Neighbouring tiles must share a habitat type.")
+            return false
+        } else {
+            return true
+        }
+    }
```

Comment:
Love that you're using stuff like `[].some`.

This could be simplified and more efficient though.

Because we know what the return value of this function should be as soon as we find an invalid tile, we can do an early return. 

```suggestion
    checkNeighbours() {
        for (const neighbour of this.neighbours) {
          if (neighbour !== null) {
            const hasMatchingHabitat = this.habitatType.some((type) => neighbour.habitatType.includes(type))
            if(!hasMatchingHabitat) {
               console.log("Invalid neighbour. Neighbouring tiles must share a habitat type.")
               return false;
            } 
          }
        }
        return true;
    }
```

Reply:
Fun alternative I thought of:

You could assert that there's a matching habitat by creating a `Set`. Because you know that a set can't have duplicate items, you'd assert that if the length of the Set is less than the length of the two habitat arrays, there is a match.

```
const combinedHabitats = new Set([...this.habitatType, ...neighbour.habitatType]);
const hasMatchingHabitat = combinedHabitats.length < this.habitatType.length + neighbour.habitatType.length
```